packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -161,15 +164,11 @@ class _FirestoreQueryBuilderState<Document>
     // "build" – most commonly ListView's itemBuilder
     Future.microtask(() => setState(() {}));
 
-    final expectedDocsCount = (_pageCount + 1) * widget.pageSize
-
-        /// The "+1" is used to voluntarily fetch one extra item,
-        /// used to determine whether there is a next page or not.
-        /// This extra item will not be rendered.
-        +
-        1;
-
-    final query = widget.query.limit(expectedDocsCount);
+    final query = (_isInitialized)
+        ? widget.query
+            .startAfterDocument(_lastQueriedDocument)
```

Comment:
Thank you for your review. From my understanding, the issue described appears to be relevant when passing data values directly within the query. In this case, we are explicitly utilizing the last queried document, thus mitigating this particular concern.

This same concern is discussed here 
https://youtu.be/poqTHxtDXwU?t=276 - at 4:32, where it's emphasised that cloud firestore leverages documentID when the Document object is passed instead of individual data values.

Having said that, I'm not sure if you are referring to this bug, which was only resolved a few years back.
https://github.com/firebase/flutterfire/issues/2936

However, to validate this scenario, I have set up a sample Firestore test database where I loaded 50 duplicate users, followed by 10 distinct users, utilizing a page size of 10.
https://github.com/a-s-k-u/FirebaseUI-Flutter/commit/fa21915eb99792a2a2dea1b21f1abade877d1213
There was an edge scenario I noted when end of documents is reached, that can be fixed with this [missing check](https://github.com/a-s-k-u/FirebaseUI-Flutter/commit/fa21915eb99792a2a2dea1b21f1abade877d1213#diff-df489d5e040c259b77c51b59326d57ffa8ebbf45b0bee42e7bc1d257ab3ea68e)

Other than that, the pagination worked as expected; and all 60 items were loaded up.

Reply:
@rrousselGit any feedback on that?

Reply:
Sounds good to me. It appears that the original concern was fixed.
I'm fine with this change then

Reply:
I'm wondering whether the data gets updated if the `_lastQueriedDocument` is deleted by a user...

Reply:
@ShahoodulHassan - Good point! I suppose since we are subscribing to that page already through .listen(), any changes ( including add or delete or update ) on that page documents will propagate back to us, and _lastQueriedDocument  will be updated accordingly. I'll try to see if I can put up a test case to confirm that. Also, want to make sure it'll not result in duplicate docs.

Reply:
@a-s-k-u so, could u find any time to test this scenario out?

Reply:
Hi @ShahoodulHassan - I did. If a user tries to delete or modify any document that is in the last queried page(where an active listener is in place), then, it is resulting in duplicate documents getting populated to the list. The other aspects of pagination does not have any impact and works as expected.
Thanks for raising it. I'll update the PR shortly so that the list is populated only for 'added' document change event.

Reply:
@a-s-k-u If my understanding is correct here, with this change what we are effectively doing is fetching documents after the '_lastQueriedDocument'. Also, our listener is now listening to changes to these set of documents only. My understanding is that - any updates to earlier fetched documents(earlier page docs) will not trigger our listener and therefore will not update UI. What do you think?

Reply:
You're absolutely right, @vaibhav891. It's a crucial trade off that we can't overlook. If it were not for _querySubscription?.cancel(), we could still keep an eye on all pages. However, in my opinion, having an excessive number of listeners just to achieve real-time functionality seems unnecessary for the vast majority of use cases. I'm more inclined to the idea of maintaining at least one listener for the topmost page as a practical solution to update the entire pagination through a callback input, in case of changes. We can clarify in the documentation that only the first page will offer real-time updates, allowing end users to adjust their queries accordingly. 
I'm looking forward to reaching a consensus on how to proceed.

Reply:
I don't think having a single listener for the current page is ideal. At least not by default.
Otherwise folks _will_ wonder why when their users scoll up in a ListView, items are no-longer updated.

We should listen to all items, but optimize reads. And maybe have a flag to listen to only a few items